homeassistant/helpers/script.py

```diff
@@ -611,6 +612,16 @@ def log_cb(level, msg):
                 task.cancel()
             remove_triggers()
 
+    async def _async_set_step(self):
+        """Set a variable value."""
+        self._log("Executing step %s", self._script.last_action)
+        new_values = self._action.get(CONF_SET, {})
+        self._log(str(self._variables))
+        for k, v in new_values.items():
+            self._log("Variable %s: %s", k, v)
+            self._variables[k] = template.render_complex(v, self._variables)
```

Comment:
That's interesting how you're rendering each field with the previous processed fields already set. I had not thought about that.

Wonder if we should do the same for `variables:`

Reply:
🤷  It was the easiest way I could imagine to get the counter example to work.

Reply:
What would it do in `variables:` though?
Wait... redefine trigger and stuff, and preprocess inputs to scripts. That's pretty neat! I vote yes!